tests/Microsoft.Identity.Test.Unit/ApiConfigTests/AuthorityTests.cs

```diff
@@ -433,6 +433,30 @@ public void IsDefaultAuthorityTest()
             Assert.IsFalse(s_b2cAuthority.AuthorityInfo.IsDefaultAuthority);
         }
 
+        [DataTestMethod]
+        [DataRow(TestConstants.AuthorityCommonTenant, typeof(AadAuthority), "Aad")]
+        [DataRow(TestConstants.AuthorityCommonPpeAuthority, typeof(AadAuthority), "Aad")]
+        [DataRow(TestConstants.AuthorityConsumersTenant, typeof(AadAuthority), "Aad")]
+        [DataRow(TestConstants.AuthorityOrganizationsTenant, typeof(AadAuthority), "Aad")]
+        [DataRow(TestConstants.AuthorityGuidTenant, typeof(AadAuthority), "Aad")]
+        [DataRow(TestConstants.DstsAuthorityCommon, typeof(DstsAuthority), "Dsts")]
+        [DataRow(TestConstants.DstsAuthorityTenantless, typeof(DstsAuthority), "Dsts")]
+        [DataRow(TestConstants.ADFSAuthority, typeof(AdfsAuthority), "Adfs")]
+        [DataRow(TestConstants.CiamAuthorityMainFormat, typeof(CiamAuthority), "Ciam")]
+        public void VerifyConfigAuthorityType(string authorityHost, Type authorityTypeInstance, string authorityType)
```

Comment:
Can you add the unit test which we discussed about as well? 

```csharp
 [TestMethod]
 public async Task TestAsync()
 {
 
     // TODO: add HttpManager mocks that serve dSTS response
    
     string AuthorityHost = "https://usnorth-passive-dsts.dsts.core.windows.net/dstsv2/";
     string tenantId = "tenant";

     UriBuilder uriBuilder = new UriBuilder(AuthorityHost)
     {
         Path = tenantId
     };

     var app = ConfidentialClientApplicationBuilder
         .Create(TestConstants.ClientId)
         .WithAuthority(AuthorityHost, tenantId)
         .WithClientSecret("secret")
         .Build();

     AuthenticationResult res = await app
         .AcquireTokenForClient(TestConstants.s_scope)
         .WithTenantIdFromAuthority(uriBuilder.Uri)
         .ExecuteAsync()
         .ConfigureAwait(false);
 }
````

Reply:
Added